content/posts/2025-02-26-whats_new_1_8.md

```diff
@@ -0,0 +1,295 @@
+3. [Egress Firewall](https://docs.openshift.com/container-platform/4.18/networking/network_security/egress_firewall/viewing-egress-firewall-ovn.html) - controls what external hosts a pod can connect to
+
+In Figure 3, the Traffic flows table shows a column for Network Events.  The first row shows a drop from a network policy and the second row allowed the packet in an admin network policy.
+
+![Flows table with Network Events]({page.image('whats-new-1-8/flows-network_events.png')})
+Figure 3: Flows table with Network Events
+
+### eBPF Flow Filter Enhancements
+
+eBPF flow filter was first introduced in Network Observability 1.6.  It lets you decide what you want to observe at the initial collector stage, so you can minimize the resources used by observability.  It comes with more enhancements in this release.
+
+#### More than one flow filter rule
+
+Instead of one filter rule, you can now have up to 16 rules, thus removing a major limitation of this feature.  The order of preference is based on the most specific match, which is the longest prefix match of the CIDR or if tied, the longest prefix match of the peer CIDR.  For example, if one rule has a CIDR of 10.0.0.0/16 and another with 10.0.1.0/24, then if the address is 10.0.1.128, it would match the second rule because that's more specific (24-bit prefix vs. 16-bit prefix).  This also implies no two rules can have the same CIDRs.
+
+#### Peer CIDR
+
+`peerCIDR` is a new option that specifies the address on the server side, which is the destination when you make a request and the source when it makes a response.  There still exists a `peerIP` that can only specify a host.  However, with `peerCIDR`, you can simply use it to specify a host (/32) or a subnetwork.
+
+In summary, use `cidr` for the client side address and `peerCIDR` for the server side address.
+
+#### Sampling rate per rule
+
+Each rule can have its own sampling rate, which gives you a lot of flexibility.   For example, the eBPF Agent can be configured to process all TCP packets with the SYN flag (sampling = 1), which ensures it doesn't miss a TCP connection.  For all other packets, set a sampling rate greater than 1 such as 50 (Listing 5).
+
+```
+spec:
+  agent:
+    type: eBPF
+    ebpf:
+      flowFilter:
+        enable: true
+        rules:
+          - action: Accept
+            protocol: TCP
+            tcpFlags: SYN
+            sampling: 1
+          - action: Accept
+            cidr: 0.0.0.0/0
+            sampling: 50
+```
+
+<p style="text-align: center">Listing 5: Flow filter with different sampling value</p>
+
+If sampling is not specified, it uses the sampling value from FlowCollector.
+
+#### Include packet drops
+
+Another new option is **pktDrops**.  With **pktDrops: true** and **action: Accept**, it includes the packet only if it's dropped.  The prerequisite is that the eBPF feature, **PacketDrop** is enabled, which requires eBFP to be in **privileged** mode.  Listing 6 shows an example configuration.
```

Comment:
There's a limitation, not sure if it has to be mentioned here: if user also enables NetworkEvents, this "pktDrops" filter will not capture packets dropped by network policy.

We discussed to introduce a new filter for NetworkEvents, but that's not done yet.

Reply:
Added the limitation